Amazon.IonDotnet/Internals/Text/IonTextRawWriter.cs

```diff
@@ -228,9 +228,19 @@ public void Write(double d)
                 return;
             }
 
-            //TODO find a better way
-            var str = d.ToString(CultureInfo.InvariantCulture);
-            _writer.Write(str);
+            String str;
+
+            // Differentiate between negative zero and zero.
+            if (d == 0 && BitConverter.DoubleToInt64Bits(d) < 0)
+            {
+                str = "-0e0";
+                _writer.Write(str);
+            }
+            else
+            {
+                str = d.ToString("R");
```

Comment:
i don't know what this is for...help me understand please?

Reply:
The "R" is the round-trip format specifier, which will ensure that a numeric value that is converted to a string is parsed back into the same numeric value. The previous implementation when converting the double to string will round the decimal digits to a certain length, so some precision will be lost there.

Reply:
Could we also add a comment in here to clarify this?